test/functional/test_framework/test_node.py

```diff
@@ -586,18 +590,21 @@ def busy_wait_for_debug_log(self, expected_msgs, timeout=60):
         yield
 
         while True:
-            found = True
+            found = []
             with open(self.debug_log_path, "rb") as dl:
                 dl.seek(prev_size)
                 log = dl.read()
 
-            for expected_msg in expected_msgs:
-                if expected_msg not in log:
-                    found = False
+            for i, expected_msg in enumerate(expected_msgs):
+                if expected_msg in log:
+                    found.append(i)
 
-            if found:
+            if len(found) == len(expected_msgs):
```

Comment:
Wouldn't this return if the same expected message appears multiple times in the log? I think we should collect the indexes into a set instead.